pymbolic/geometric_algebra/__init__.py

```diff
@@ -388,25 +425,23 @@ def orthogonal_blade_product_weight(a_bits, b_bits, space):
 
 # {{{ multivector
 
-def _cast_or_ni(obj, space):
+def _cast_or_ni(obj: Any, space: Space) -> MultiVector:
     if isinstance(obj, MultiVector):
         return obj
     else:
         return MultiVector(obj, space)
 
 
-class MultiVector:
+@expr_dataclass(init=False, hash=False)
+class MultiVector(Generic[CoeffT]):
```

Comment:
This isn't an expression though?

EDIT: read lower down that this is very intentional, so fair enough!